File: 백준/Silver/11723. 집합/집합.java
import java.io.*;
import java.util.*;

/**
 * 20개의 상태를 나타내기 위해 INT형 4바이트 20개를 사용하는 건 낭비일 수 있다.
 * 비트마스킹을 이용하여 각 비트가 하나의 상태를 나타내도록 하자.
 */

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int m = Integer.parseInt(br.readLine());

        StringBuilder sb = new StringBuilder();
        int state = 0; // 숫자 20개가 INT 32비트 중 하위 20개를 사용하여 각 비트를 나타냄
        while (m-- > 0) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            String command = st.nextToken();
            if (command.equals("add")) {
                int num = Integer.parseInt(st.nextToken());
                state |= (1 << (num - 1));
            } else if (command.equals("remove")) {
                int num = Integer.parseInt(st.nextToken());
                state &= (~(1 << (num - 1)));
            } else if (command.equals("check")) {
                int num = Integer.parseInt(st.nextToken());
                sb.append((state >> (num - 1)) & 1).append('\n');
            } else if (command.equals("toggle")) {
                int num = Integer.parseInt(st.nextToken());
                state ^= (1 << (num - 1));
            } else if (command.equals("all")) {
                state = 0xfffff;
            } else {
                state = 0;
            }
        }
        System.out.print(sb);
    }
}

File: 백준/Silver/11723. 집합/집합.kt
import java.util.*

// i 번째 비트 키기
infix fun Int.on(i: Int) = this or (1 shl i - 1)

// i 번째 비트 끄기
infix fun Int.off(i: Int) = this and (1 shl i - 1).inv()

// i 번째 비트 체크하기
infix fun Int.check(i: Int) = if (this and (1 shl i - 1) >= 1) 1 else 0

// i 번째 비트 토글하기
infix fun Int.toggle(i: Int) = this xor (1 shl i - 1)

fun main() = with(System.`in`.bufferedReader()) {
    var status = 0
    val sb = StringBuilder()
    repeat(readLine().toInt()) {
        val parts = readLine().split(" ")
        val command = parts[0]
        val num = parts.last().toIntOrNull() ?: 0
        when (command) {
            "add" -> status = status on num
            "remove" -> status = status off num
            "check" -> sb.append(status check num).append('\n')
            "toggle" -> status = status toggle num
            "all" -> status = (1 shl 20) - 1
            "empty" -> status = 0
        }
    }
    print(sb)
}